### PR TITLE
Refactor tracing demo to show A->B->C call stack.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Refactor tracing demo to show A->B->C call stack.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/2504

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

File: examples/src/main/java/io/dapr/examples/state/StateClient.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public String toString() {
     }
   }
 
-  private static final String STATE_STORE_NAME = "statestore";
+  private static final String STATE_STORE_NAME = "state/store";
```
Comment (yaron2) [inline examples/src/main/java/io/dapr/examples/state/StateClient.java:37]: Is this intentional? `state/store` instead of `statestore`

Comment (artursouza) [reply examples/src/main/java/io/dapr/examples/state/StateClient.java:37]: haha. It was a left over from another work.

Comment (yaron2) [reply examples/src/main/java/io/dapr/examples/state/StateClient.java:37]: happens :)

**Review by yaron2 — APPROVED** [review]
LGTM

**Pull request merged**